### DISCUSSION TITLE
expo plugin for the messaging package

### DISCUSSION THREAD

**Discussion by Stevemoretz** [invertase/react-native-firebase#5885] [Ideas] [answered]
This is a really useful package, I used to use it on bare workflow but had to migrate, now with eas it is possible to use this, but needs a plugin, I notice this particular package doesn't have a plugin, is there any plan for this? Thank you so much.

By the way I'm talking about this package : https://github.com/invertase/react-native-firebase/tree/main/packages/messaging

**Comment by mikehardy** [comment, marked as answer]
Hi there! I don't believe it needs one, reference: https://github.com/invertase/react-native-firebase/pull/5480#issuecomment-933164785

**Reply by Stevemoretz** [reply]
Hello there! Thank you for the reply, however I don't quite follow, in that comment I think there are two things referenced.

1. Use of expo-notification

I know this is a great package, in fact I'm using it but it's not really related to my request.

2. Use of this package https://github.com/barthap/with-rn-firebase

this clearly mentions : `Since React Native Firebase 12.4.0 this plugin is deprecated in favor of built-in plugins. Learn more here.` and the learn more here link takes us back to the documentation of rn-firebase, so I encountered an infinite loop here lol.

Unless you meant something else, that I did not realize.

**Reply by mikehardy** [reply]
I mean specifically that the messaging package here has no configuration, except to configure the notification icon, as far as I know.
Which is achievable via expo-notification plugin, if I understand that comment correctly

If you examine our usage doc you'll notice there is no necessary native code, with exception of notification icon on android https://rnfirebase.io/messaging/usage#notification-color

So I think there is no plugin needed

**Reply by Stevemoretz** [reply]
Thank you so much makes total sense, I'm new to eas didn't figure to check that, now I realize I should check configuration of each package, if it only needs pod install then it's handled automatically. Maybe add that to the documentation for the doofus people like me? Lol

**Reply by mikehardy** [reply]
Well, you may find that you want some stuff in firebase.json, but that's like a big react-native-firebase config plugin :-). Good luck with your project
